src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
+  async githubAuthCallback(@Query('code') code: string) {
+    try {
+      const jwtToken = await this.authService.handleGitHubCallback(code);
+
+      return jwtToken;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Post('logout')
+  async logout(@Req() req: TRequest) {
+    try {
+      return await this.authService.logout(req);
+    } catch (error) {
+      if (error.status)
+        throw new HttpException(error.message, error.status);
+      else throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+    }
```

Comment:
PullProofAI

The `logout` method also suffers from similar issues with error handling. Ensure consistent error handling and response generation for better API behavior.

```diff
+       try {
+           return await this.authService.logout(req);
+       } catch (error) {
+           if (error.status)
+               throw new HttpException(error.message, error.status);
+           else
+               throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+       }
```

#### Overall Comments
- Ensure consistent error handling practices throughout the controller methods.
- Consider extracting common error handling logic into reusable functions to avoid code duplication.
- Validate input parameters and ensure proper validation is in place for security and data integrity.
- Make sure to follow SOLID principles for better code organization and maintainability.

Please address these points to enhance the reliability and robustness of the authentication controller.


<!-- This is an auto-generated comment by PullProofAI -->